src/components/ErrorForm/ErrorFormWithList.jsx

```diff
@@ -1,62 +1,49 @@
-import React from 'react'
+import React, { useRef, useState } from 'react'
 import formStyle from './errorForm.module.css'
 
-class ErrorFormWithList extends React.Component {
-  emailRef = React.createRef()
+export default function ErrorFormWithList() {
+  const emailRef = useRef(null)
+  const [emailInputValid, setEmailInputValid] = useState(false)
 
-  constructor(props) {
-    super(props)
-
-    this.state = {
-      emailInputValid: undefined,
-    }
-    this.handleSubmit = this.handleSubmit.bind(this)
-  }
-
-  handleSubmit(e) {
-    const value = this.emailRef.current.value
-    const emailInputValid = value !== ''
-    this.setState({ emailInputValid })
+  const handleSubmit = e => {
+    const value = emailRef.current.value
```

Comment:
wee destructure opportunity here too (sorry I know this is my code 😂)

```suggestion
    const { value } = emailRef.current
```